sklearnex/preview/linear_model/ridge.py

```diff
@@ -0,0 +1,343 @@
+            [
+                (n_samples > 0, "Number of samples is less than 1."),
+                (not issparse(*data), "Sparse input is not supported."),
+                (not model_is_sparse, "Sparse coefficients are not supported."),
+            ]
+        )
+        if not dal_ready:
+            return patching_status
+
+        patching_status.and_condition(
+            self._test_type_and_finiteness(*data), "Input X is not supported."
+        )
+
+        return patching_status
+
+    def _onedal_gpu_supported(self, method_name, *data):
+        if method_name == "fit":
+            return self._onedal_fit_supported("gpu", method_name, *data)
+        if method_name == "predict":
+            return self._onedal_predict_supported(method_name, *data)
+        raise RuntimeError(f"Unknown method {method_name} in {self.__class__.__name__}")
+
+    def _onedal_cpu_supported(self, method_name, *data):
+        if method_name == "fit":
+            return self._onedal_fit_supported("cpu", method_name, *data)
+        if method_name == "predict":
+            return self._onedal_predict_supported(method_name, *data)
+        raise RuntimeError(f"Unknown method {method_name} in {self.__class__.__name__}")
+
+    def _initialize_onedal_estimator(self):
+        onedal_params = {
+            "fit_intercept": self.fit_intercept,
+            "alpha": self.alpha,
+            "copy_X": self.copy_X,
+        }
+        self._onedal_estimator = onedal_RidgeRegression(**onedal_params)
+
+    def _daal_fit(self, X, y, sample_weight=None):
+        daal4py_fit_ridge(self, X, y, sample_weight)
+        self._onedal_estimator.n_features_in_ = _num_features(X, fallback_1d=True)
+        self._onedal_estimator.coef_ = self.coef_
+        self._onedal_estimator.intercept_ = self.intercept_
+
+    def _onedal_fit(self, X, y, sample_weight, queue=None):
+        assert sample_weight is None
+
+        check_params = {
+            "X": X,
+            "y": y,
+            "dtype": [np.float64, np.float32],
+            "accept_sparse": ["csr", "csc", "coo"],
+            "y_numeric": True,
+            "multi_output": True,
+            "force_all_finite": False,
+        }
+        if sklearn_check_version("1.2"):
+            X, y = self._validate_data(**check_params)
+        else:
+            X, y = check_X_y(**check_params)
+
+        if sklearn_check_version("1.0") and not sklearn_check_version("1.2"):
+            self._normalize = _deprecate_normalize(
+                self.normalize,
+                default=False,
+                estimator_name=self.__class__.__name__,
+            )
+
+        self._initialize_onedal_estimator()
+        try:
+            # Falling back to daal4py if the device is CPU since
+            # onedal does not support non-scalars for alpha, thus
+            # should only be used for GPU to not limit the functionality
+            cpu_device = queue is None or queue.sycl_device.is_cpu
+            if cpu_device:
```

Comment:
I know its all going to the same place, but this causes the cpu branch of the oneDAL LinearRegression never to run via sklearnex. I sorta thought there would be an alpha check in the if statement...